devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java

```diff
@@ -237,6 +244,54 @@ public void startDev() {
                 resources.append(file.getAbsolutePath());
                 res = file.getAbsolutePath();
             }
+
+            final Configuration compileCp = project.getConfigurations().getByName("compileClasspath");
+            final DependencySet compileCpDependencies = compileCp.getAllDependencies();
+
+            //Build a set of paths to all the projects the application depends on within
+            //the local workspace.
+            Set<String> localProjectDependencies = new HashSet<>();
+            for (Dependency dependency : compileCpDependencies) {
+                if (!(dependency instanceof ProjectDependency)) {
+                    continue;
+                }
+
+                Project dependencyProject = ((ProjectDependency) dependency).getDependencyProject();
+                localProjectDependencies.add(dependencyProject.getPath());
```

Comment:
Having obtained `dependencyProject` is there a reason you are not processing it right away? Instead of collecting a set and then iterating through all the projects filtering out those that aren't deps?

Reply:
Doh, no reason for that. I originally didn't have a `Project` to work with, and after getting it from the `ProjectDependency` I forgot to factor out the first iteration that built the set.

My mistake, v3 incoming shortly!